diablo/jobs/dblink_to_redshift_job.py

```diff
@@ -30,26 +33,111 @@
+        #  3. We collapse the names of the three section into a single name/title for section 123
+
+        # IMPORTANT: These will be ordered by schedule (time and location)
+        sql = f"""
+            SELECT
+                section_id,
+                trim(concat(
+                    meeting_days,
+                    meeting_end_date,
+                    meeting_end_time,
+                    meeting_location,
+                    meeting_start_date,
+                    meeting_start_time
+                )) as schedule
+            FROM sis_sections
+            WHERE term_id = :term_id
+                AND meeting_days <> ''
+                AND meeting_end_date <> ''
+                AND meeting_end_time <> ''
+                AND meeting_location <> ''
+                AND meeting_start_date <> ''
+                AND meeting_start_time <> ''
+            ORDER BY schedule
+        """
+        rows = db.session.execute(
+            text(sql),
+            {
+                'term_id': term_id,
+            },
+        )
+        cross_listings = {}
+        previous_schedule = None
+        current_key = None
+
+        for row in rows:
+            section_id = row['section_id']
+            schedule = row['schedule']
+            if section_id not in cross_listings:
+                if schedule != previous_schedule:
+                    current_key = section_id
+                    cross_listings[current_key] = []
+                else:
+                    cross_listings[current_key].append(section_id)
+            previous_schedule = schedule
+
+        # Toss out section_ids with no cross-listings
+        for section_id, section_ids in cross_listings.copy().items():
+            if not section_ids:
+                cross_listings.pop(section_id)
+
+        # Delete existing rows
+        db.session.execute(CrossListing.__table__.delete().where(CrossListing.term_id == term_id))
+
+        def chunks(data, chunk_size=500):
+            iterator = iter(data)
+            for i in range(0, len(data), chunk_size):
+                yield {k: data[k] for k in islice(iterator, chunk_size)}
+
+        delete_section_ids = []
+
+        for cross_listings_chunk in chunks(cross_listings):
+            cross_listing_count = len(cross_listings_chunk)
+            query = 'INSERT INTO cross_listings (term_id, section_id, cross_listed_section_ids, created_at) VALUES'
+            for index, (section_id, cross_listed_section_ids) in enumerate(cross_listings_chunk.items()):
+                query += f' (:term_id, {section_id}, ' + "'{" + _join(cross_listed_section_ids, ', ') + "}', now())"
+                if index < cross_listing_count - 1:
+                    query += ','
+                # Cross-referenced section ids will be deleted in  sis_sections table
+                delete_section_ids.extend(cross_listed_section_ids)
+            db.session.execute(query, {'term_id': term_id})
+
+        # Remove section ids that a
```

Comment:
@johncrossman Trivial, but did this comment get cut off?

Reply:
Thank you @lyttam  Amending.